### DISCUSSION TITLE
New Pydantic Models query feature behaves strangely with nested models

### DISCUSSION THREAD

**Discussion by ollz272** [fastapi/fastapi#12214] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from idlelib.query import Query
from typing import Annotated
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()

class NestedModel(BaseModel):
    foo: str


class FilterParams(BaseModel):
    bar: NestedModel


@app.get("/items/")
async def read_items(filter_query: Annotated[FilterParams, Query()]):
    return filter_query
```

### Description

The endpoint seems to behave in weird ways.
Firstly, when looking at the docs, it seems to (correctly? i think?) say that bar is an object which takes `foo` as a field. Great.
![Screenshot 2024-09-18 at 14 42 43](https://github.com/user-attachments/assets/dcbfbbf5-645a-4034-88e5-d2c402b811af)
When using the try it out feature though, it suggests this as the request:
`http://127.0.0.1:8000/items/?foo=string`

Which results in:
```json
{
  "detail": [
    {
      "loc": [
        "query",
        "bar"
      ],
      "msg": "Field required",
      "type": "missing"
    }
  ]
}
```

Which seems odd that fast api docs suggested this? i think?

When trying to make a more sensible (i think?) request:

```http://127.0.0.1:8000/items/?bar={%22foo%22:%22str%22}```

We get:
```json
{
  "detail": [
    {
      "loc": [
        "query",
        "bar"
      ],
      "msg": "Input should be a valid dictionary or object to extract fields from",
      "type": "model_attributes_type"
    }
  ]
}
```

Which again... weird?

I'm not sure what the use case is for nested models as query params, but fastapi should probably:
1. Figure out a way to support this
2. Try to stop this as it doesn't really make sense

### Operating System

macOS

### FastAPI Version

0.115.0

### Pydantic Version

2.9.2

### Python Version

Python 3.12.4

**Comment by jjfantini** [comment]
Did you find a solution?
